File: src/main/java/com/manning/simplysend/repositories/ItemsRepository.java
package com.manning.simplysend.repositories;

import com.manning.simplysend.entities.Item;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;

public interface ItemsRepository extends PagingAndSortingRepository<Item, Long> {
    Page<Item> findAllByType(String type, Pageable pageable);
}

File: src/main/kotlin/com/manning/simplysend/repositories/ItemsRepository.kt
package com.manning.simplysend.repositories

import com.manning.simplysend.entities.Item
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository

interface ItemsRepository : PagingAndSortingRepository<Item, Long> {

    fun findAllByType(type: String, pageable: Pageable): Page<Item>
}